stable/jenkins/README.md

```diff
@@ -25,6 +25,80 @@ $ helm install my-release stable/jenkins
 
 A major chart version change (like v0.40.0 -> v1.0.0) indicates that there is an incompatible breaking change needing manual actions.
 
+### 2.0.0 Configuration as Code now default + container does not run as root anymore
+
+#### Configuration as Code new default
+
+Configuration is done via [Jenkins Configuration as Code Plugin](https://github.com/jenkinsci/configuration-as-code-plugin) by default.
+That means that changes in values which result in a configuration change are always applied.
+In contrast the XML configuration was only applied during the first start and never altered.
+
+:exclamation::exclamation::exclamation: Attention: This also means if you manually altered configuration then this will most likely be reset to what was configured by default. :exclamation::exclamation::exclamation:
+
+`securityRealm` and `authorizationStrategy` are not configured using configuration as code so these should be identical to what you configured before.
```

Comment:
I did not come up with proper defaults for `securityRealm` and `authorizationStrategy`. Both are configured via XML. I guess that a huge percentage of users of this chart re-configured them. So providing a new default here would reset their changes and possibly lock them out.

Adding native support for this via `master.JCasC.authorizationStrategy` and `master.JCasC.securityRealm` could be an option. But then we should at least add a note that in case they already configured it via master.JCasC.configScripts then they would need to move the configuration.

Adding support for jobs is easier. We could do what you propose and also include an example in the README. Would you mind creating a PR for that?

I would not mind keeping this PR open for a while. I opened it by intention early to get feedback like yours abount things which are missing. Ideally the PR for 2.0.0 just contains changes in values.yaml so that it's easy to opt out and move all other necessary changes to seperate PRs which could be merged before this one.

Reply:
#22408 added `master.JCasC.securityRealm` and `master.JCasC.authorizationStrategy`

At this point, I'm thinking converting jobs to JCasC can wait until post 2.0. It will likely require adding the `job-dsl` plugin.

Reply:
agree